src/System.Slices/System/ReadOnlySpan.cs

```diff
@@ -143,6 +143,16 @@ internal ReadOnlySpan(object obj, UIntPtr offset, int length)
             return new ReadOnlySpan<T>(slice.Object, slice.Offset, slice.Length);
         }
 
+        /// <summary>
+        /// Gets ArraySegment containing the ReadOnlySpan's data.
+        /// </summary>
+        /// <remarks>This method can be used for interop, as array data is mutable, an array is allocated and copy of the data is returned in it.</remarks>
+        public static explicit operator ArraySegment<T>(ReadOnlySpan<T> span)
```

Comment:
For my money, I'd want it to be *shockingly obvious* that  I'm allocating here; by comparison to `CreateArray`, I wonder whether  `CreateArraySegment()` is a better thing here